File: class-comparer/create_sample_files.py
import openpyxl
from openpyxl import Workbook

# Create Manual XLSX file
wb_manual = Workbook()
ws_manual = wb_manual.active
ws_manual.title = "Classes"

# Add headers
ws_manual['A1'] = 'Story'
ws_manual['B1'] = 'Class'

# Sample data for manually created classes
manual_data = [
    ('User Authentication', 'User'),
    ('User Authentication', 'AuthenticationService'),
    ('User Authentication', 'LoginController'),
    ('User Authentication', 'PasswordValidator'),
    ('Shopping Cart', 'Cart'),
    ('Shopping Cart', 'CartItem'),
    ('Shopping Cart', 'CartService'),
    ('Shopping Cart', 'PriceCalculator'),
    ('Payment Processing', 'Payment'),
    ('Payment Processing', 'PaymentGateway'),
    ('Payment Processing', 'Transaction'),
    ('Payment Processing', 'PaymentValidator'),
    ('Order Management', 'Order'),
    ('Order Management', 'OrderService'),
    ('Order Management', 'OrderRepository'),
    ('Order Management', 'OrderStatus'),
]

# Add data rows
for idx, (story, class_name) in enumerate(manual_data, start=2):
    ws_manual[f'A{idx}'] = story
    ws_manual[f'B{idx}'] = class_name

# Save manual file
wb_manual.save('manual_classes.xlsx')
print("Created: manual_classes.xlsx")

# Create LLM-generated XLSX file
wb_llm = Workbook()
ws_llm = wb_llm.active
ws_llm.title = "Classes"

# Add headers
ws_llm['A1'] = 'Story'
ws_llm['B1'] = 'Class'

# Sample data for LLM-generated classes (some overlap, some different)
llm_data = [
    ('User Authentication', 'User'),
    ('User Authentication', 'AuthenticationService'),
    ('User Authentication', 'LoginController'),
    ('User Authentication', 'TokenGenerator'),  # Different from manual
    ('User Authentication', 'SessionManager'),   # Different from manual
    ('Shopping Cart', 'Cart'),
    ('Shopping Cart', 'CartItem'),
    ('Shopping Cart', 'CartService'),
    ('Shopping Cart', 'DiscountCalculator'),     # Different from manual
    ('Payment Processing', 'Payment'),
    ('Payment Processing', 'PaymentGateway'),
    ('Payment Processing', 'Transaction'),
    ('Payment Processing', 'CreditCardProcessor'), # Different from manual
    ('Order Management', 'Order'),
    ('Order Management', 'OrderService'),
    ('Order Management', 'OrderRepository'),
    ('Order Management', 'Shipment'),            # Different from manual
    ('Order Management', 'Invoice'),             # Different from manual
]

# Add data rows
for idx, (story, class_name) in enumerate(llm_data, start=2):
    ws_llm[f'A{idx}'] = story
    ws_llm[f'B{idx}'] = class_name

# Save LLM file
wb_llm.save('llm_classes.xlsx')
print("Created: llm_classes.xlsx")

print("\nSample files created successfully!")
print("- manual_classes.xlsx: 16 classes across 4 stories")
print("- llm_classes.xlsx: 18 classes across 4 stories")

File: class-comparer/create_sample_files.js
const XLSX = require('xlsx');

// Create Manual workbook
const manualData = [
    ['Story', 'Class'],
    ['User Authentication', 'User'],
    ['User Authentication', 'AuthenticationService'],
    ['User Authentication', 'LoginController'],
    ['User Authentication', 'PasswordValidator'],
    ['Shopping Cart', 'Cart'],
    ['Shopping Cart', 'CartItem'],
    ['Shopping Cart', 'CartService'],
    ['Shopping Cart', 'PriceCalculator'],
    ['Payment Processing', 'Payment'],
    ['Payment Processing', 'PaymentGateway'],
    ['Payment Processing', 'Transaction'],
    ['Payment Processing', 'PaymentValidator'],
    ['Order Management', 'Order'],
    ['Order Management', 'OrderService'],
    ['Order Management', 'OrderRepository'],
    ['Order Management', 'OrderStatus'],
];

const manualWS = XLSX.utils.aoa_to_sheet(manualData);
const manualWB = XLSX.utils.book_new();
XLSX.utils.book_append_sheet(manualWB, manualWS, 'Classes');
XLSX.writeFile(manualWB, 'manual_classes.xlsx');
console.log('Created: manual_classes.xlsx');

// Create LLM workbook
const llmData = [
    ['Story', 'Class'],
    ['User Authentication', 'User'],
    ['User Authentication', 'AuthenticationService'],
    ['User Authentication', 'LoginController'],
    ['User Authentication', 'TokenGenerator'],
    ['User Authentication', 'SessionManager'],
    ['Shopping Cart', 'Cart'],
    ['Shopping Cart', 'CartItem'],
    ['Shopping Cart', 'CartService'],
    ['Shopping Cart', 'DiscountCalculator'],
    ['Payment Processing', 'Payment'],
    ['Payment Processing', 'PaymentGateway'],
    ['Payment Processing', 'Transaction'],
    ['Payment Processing', 'CreditCardProcessor'],
    ['Order Management', 'Order'],
    ['Order Management', 'OrderService'],
    ['Order Management', 'OrderRepository'],
    ['Order Management', 'Shipment'],
    ['Order Management', 'Invoice'],
];

const llmWS = XLSX.utils.aoa_to_sheet(llmData);
const llmWB = XLSX.utils.book_new();
XLSX.utils.book_append_sheet(llmWB, llmWS, 'Classes');
XLSX.writeFile(llmWB, 'llm_classes.xlsx');
console.log('Created: llm_classes.xlsx');

console.log('\nSample files created successfully!');
console.log('- manual_classes.xlsx: 16 classes across 4 stories');
console.log('- llm_classes.xlsx: 18 classes across 4 stories');
console.log('\nExpected comparison results:');
console.log('  Common classes: User, AuthenticationService, LoginController, Cart, CartItem, CartService, Payment, PaymentGateway, Transaction, Order, OrderService, OrderRepository');
console.log('  Only in Manual: PasswordValidator, PriceCalculator, PaymentValidator, OrderStatus');
console.log('  Only in LLM: TokenGenerator, SessionManager, DiscountCalculator, CreditCardProcessor, Shipment, Invoice');
